DOCS-2541: Move component APIs (#3504)
viamrobotics/docs: docs/appendix/apis/_index.md

---

> Every Viam {{< glossary_tooltip term_id="resource" text="resource" >}} exposes an [Application Programming Interface (API)](https://en.wikipedia.org/wiki/API) described through [protocol buffers](https://developers.google.com/protocol-buffers).

Reviewer: ```suggestion
Every Viam {{< glossary_tooltip term_id="resource" text="resource" >}} exposes an [application programming interface (API)](https://en.wikipedia.org/wiki/API) described through [protocol buffers](https://developers.google.com/protocol-buffers).
```
[out-of-diff nit]

The API methods provided by the SDKs for each of these resource APIs wrap gRPC client requests to the machine when you execute your program, providing you a convenient interface for accessing information about and controlling the {{< glossary_tooltip term_id="resource" text="resources" >}} you have [configured](/configure/) on your machine.

> ## Machine and Fleet APIs

Author: ```suggestion
## Machine and fleet APIs
```

> ## Machine and fleet APIs

Reviewer: "Machine API" might be confusing since you use component APIs to control machines. I know what you mean and I can't think of a better heading but if you have an idea maybe change.

Author: ```suggestion
## Platform APIs
```

{{< cards >}}
{{% manualcard link="/appendix/apis/fleet/" title="Fleet Management API" %}}

> Create and manage organizations, locations, and individual machines, fragments and permissions.

Reviewer: [opt] It is a bit unclear why I can do things with individual machines both through the fleet mgmnt api as well as with the machine api. Maybe make it more clear that fleet mgmnt has to do with the app client vs machine mgmnt is more local or low-level (if that's the case)?

Author: ```suggestion
Create and manage organizations, locations, and machines, get logs from individual machines, and manage fragments and permissions.
```

{{% /manualcard %}}
{{% manualcard link="/appendix/apis/data-client/" title="Data Client API" %}}

Upload, download, filter, tag or perform other tasks on data like images or sensor readings.

{{% /manualcard %}}
{{% manualcard link="/appendix/apis/robot/" title="Machine Management API" %}}

Manage your machines: connect to your machine, retrieve status information, and send commands remotely.

{{% /manualcard %}}
{{% manualcard link="/appendix/apis/ml-training-client/" title="ML Training Client API" %}}

Submit and manage ML training jobs running on the Viam app.

{{% /manualcard %}}
{{% manualcard link="/appendix/apis/billing-client/" title="Billing Client API" %}}

Retrieve billing information from the Viam app.

{{% /manualcard %}}

{{< /cards >}}

## Component APIs

> These APIs provide interfaces for controlling and getting information from various components of a machine:

Reviewer: ```suggestion
These APIs provide interfaces for controlling and getting information from the {{< glossary_tooltip term_id="component" text="components" >}} of a machine:
```

{{< cards >}}
{{< card link="/appendix/apis/components/arm/" noimage="True" >}}
{{< card link="/appendix/apis/components/base/" noimage="True" >}}
{{< card link="/appendix/apis/components/board/" noimage="True" >}}
{{< card link="/appendix/apis/components/camera/" noimage="True" >}}
{{< card link="/appendix/apis/components/encoder/" noimage="True" >}}
{{< card link="/appendix/apis/components/gantry/" noimage="True" >}}
{{< card link="/appendix/apis/components/generic/" noimage="True" >}}
{{< card link="/appendix/apis/components/gripper/" noimage="True" >}}
{{< card link="/appendix/apis/components/input-controller/" noimage="True" >}}
{{< card link="/appendix/apis/components/motor/" noimage="True" >}}
{{< card link="/appendix/apis/components/movement-sensor/" noimage="True" >}}
{{< card link="/appendix/apis/components/power-sensor/" noimage="True" >}}
{{< card link="/appendix/apis/components/sensor/" noimage="True" >}}
{{< card link="/appendix/apis/components/servo/" noimage="True" >}}
{{< /cards >}}

## Service APIs

These APIs provide interfaces for controlling and getting information from the services you configured on a machine.

> <!-- TODO: move to APIS folder and change links -->

Author: yes, the services aren't part of this PR
